src/goto-symex/symex_assign.cpp

```diff
@@ -350,6 +358,92 @@ void symex_assignt::assign_from_struct(
   }
 }
 
+void symex_assignt::assign_constants(
+  const ssa_exprt &length,
+  const constant_exprt &new_length,
+  const ssa_exprt &char_array,
+  const array_exprt &new_char_array)
+{
+  // assign length of string
+  assign_symbol(
```

Comment:
since you know the rhs is not a `struct` you can call directly `assign_non_struct_symbol` :strawberry: